### PR TITLE
Fixes rounding to satisfy min sell amount > min tradable amount.

### PR DESCRIPTION
**Pull request by marcovc**
The rounding procedure was lacking a test for making sure that the (potentially) decreased sell amount does not violate the min tradable amount constraint.

### CODE REVIEW

File: src/core/round.py [hunk 1]
```diff
@@ -123,6 +123,60 @@ def compute_spanning_order_arborescence(orders, fee):
...
+
+        new_sell_amount = order.with_buy_amount(order.buy_amount + buy_amount_delta)\
+            .get_sell_amount_from_buy_amount(prices, fee, IntegerTraits)
+
+        # Skip order if rounding would lead to violation of max sell amount.
+        if new_sell_amount > order.max_sell_amount:
+            continue
+
+        # Skip order if rounding would lead to violation of min tradable amount.
+        if new_sell_amount < Config.MIN_TRADABLE_AMOUNT:
```
Comment (josojo) [inline src/core/round.py:154]: If we are increasing the buyAmount, then also the sell_amount must have increased. I am not sure whether this is really needed.

Comment (marcovc) [reply src/core/round.py:154]: My reasoning was: The token balance, which we need to set to zero, is a difference between two terms A - B (A is the sum of sold amount, and B the sum of a bought amount). So even if you only round down (which is what we do before the first iteration) the token balance due to rounding can be positive or negative depending if "more" rounding happens on A or B. This means that the new buy amount can be increased or decreased.

Comment (josojo) [reply src/core/round.py:154]: Ah, sorry, I thought that the token_balances[leaf_token] could only be positive.

Comment (marcovc) [reply src/core/round.py:154]: Maybe an example helps: Say that A=5/3 + 4/3, and B=2 + 1. Before rounding, in the rational world, we have A=B. If we round everything down we have A=floor(5/3)+floor(4/3)=2, and B=3. So the token balance is -1. But if we switch A with B we would get 1.

**Pull request merged**